H-913: Add non-exhaustive outline of unpublished libraries (#3383)
hashintel/hash: libs/README.md

| [@hashintel/design-system]       | TypeScript  | [npm](https://www.npmjs.com/package/@hashintel/design-system)       | To be written | A collection of styleguide-aligned reusable UI primitives for [HASH] and our [hash.ai] website       |
| [@hashintel/block-design-system] | TypeScript  | [npm](https://www.npmjs.com/package/@hashintel/block-design-system) | To be written | A relatively unopinionated set of reusable UI primitives for use in building [Block Protocol] blocks |

> ## Internal unpublished libraries
> 

Author: Maybe we don't need that heading here?
```suggestion
```
Or maybe making it smaller?
```suggestion
# Internal unpublished libraries

```

Reviewer: I'd be happy with...

```suggestion
## Internal libraries

```

...the text beneath makes clear they are unpublished. I think the key thing to highlight anyway is that these are internal, and not intended to be relied on more widely.

Author: The section above is already called "Internal libraries" 😄

> The packages inside of [`@local`](./@local) are libraries used inside the repository and are not published to package managers. all of these libraries may be subject to breaking changes. External consumers should be especially careful when using or upgrading these.

Reviewer: ```suggestion
The packages inside of [`@local`](./@local) are libraries used inside this repository which are **not** published to package managers. All of these libraries may be subject to breaking changes. External consumers should be especially careful when using or upgrading these.
```

The following list is a non-exhaustive list of packages in `@local`:

| Package                                                         | Language(s) | Docs URL      | Description                                                                                                       |
| --------------------------------------------------------------- | ----------- | ------------- | ----------------------------------------------------------------------------------------------------------------- |
| [@local/codec-rs](@local/codec)                                 | Rust        | Not hosted    | Implementation of different `serde` or byte codes used in HASH                                                    |
| [@local/hash-authorization-rs](@local/hash-authorization)       | Rust        | Not hosted    | Provides the Authorization interface and logic used in the Graph                                                  |

> | [@local/hash-graph-client](@local/hash-graph-client/typescript) | Typescript  | To be written | A generator to create a TypeScript/JavaScript client for the Graph API                                            |

Reviewer: ```suggestion
| [@local/hash-graph-client](@local/hash-graph-client/typescript) | TypeScript  | To be written | A generator to create a TypeScript/JavaScript client for the Graph API                                            |
```
